spec/models/component_spec.rb

```diff
@@ -1,141 +1,135 @@
+        component: msa_component,
       )
       ReplaceEncryptionCertificateEvent.create(
-        component: component,
+        component: msa_component,
         encryption_certificate_id: event.certificate.id
       )
       event
     end
+    let!(:upload_encryption_event_2) do
+      event = UploadCertificateEvent.create(
+        usage: CONSTANTS::ENCRYPTION,
+        value: root.generate_encoded_cert(expires_in: 2.months),
+        component: sp_component,
+      )
+      ReplaceEncryptionCertificateEvent.create(
+        component: sp_component,
+        encryption_certificate_id: event.certificate.id
+      )
+      event
+    end
+    let(:msa_service) { create(:service, entity_id: 'https://old-and-boring') }
+    let(:sp_service) { create(:service, entity_id: 'https://new-hotness') }
 
-    it 'is an MSA component with signing and encryption certs' do
-      component.services << msa_service
-      signing1 = upload_signing_certificate_event_1
-      signing2 = upload_signing_certificate_event_2
-      encryption = upload_encryption_event
-      actual_config = MsaComponent.to_service_metadata(event_id, published_at)
-
-      expected_config = {
-        published_at: published_at,
-        event_id: event_id,
-        connected_services: [{
-          entity_id: 'https://not-a-real-entity-id',
-          service_provider_id: nil
-        }],
-        matching_service_adapters: [{
-          name: component_name,
-          entity_id: entity_id,
-          encryption_certificate: {
-            name: encryption.certificate.x509.subject.to_s,
-            value: encryption.certificate.value
-          },
-          signing_certificates: [{
-            name: signing1.certificate.x509.subject.to_s,
-            value: signing1.certificate.value
-          }, {
-            name: signing2.certificate.x509.subject.to_s,
-            value: signing2.certificate.value
-          }]
-        }],
-        service_providers: []
-      }
+    it "publishes all the components and services metadata correctly" do
+      msa_component.services << msa_service
+      sp_component.services << sp_service
+      event_id = Event.last.id
+      actual_config = Component.to_service_metadata(event_id, published_at)
 
-      expect(expected_config).to eq(actual_config)
+      expect(expected_config(event_id)).to eq(actual_config)
     end
 
-    it 'is an SP component with a service' do
-      sp_component.services << sp_service
-      actual_config = SpComponent.to_service_metadata(event_id, published_at)
-      expected_config = {
+    def expected_config(event_id)
+      {
         published_at: published_at,
         event_id: event_id,
         connected_services: [
           {
-            entity_id: "https://not-a-real-entity-id",
+            entity_id: sp_service.entity_id,
             service_provider_id: sp_component.id
+          },
+          {
+            entity_id: msa_service.entity_id,
+            service_provider_id: nil
```

Comment:
I thought we were not allowing this scenario? Or is it just a test scenario?

Reply:
@jakubmiarka I think I need a pretty picture 😛 

Have updated